src/sortStudents.ts

```diff
@@ -1,16 +1,55 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  function setOrder(): number {
+    return order === 'asc' ? 1 : -1;
+  }
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  type Compare = string | number | boolean;
+
+  function compare(x: Student): Compare {
+    switch (sortBy) {
+      case 'name':
+      case 'surname':
+        return x[sortBy].toUpperCase();
+      case 'age':
+      case 'married':
+        return x[sortBy];
+      case 'averageGrade':
+      default:
+        return x.grades.reduce((a, b) => a + b, 0) / x.grades.length;
+    }
+  }
+
+  return students.slice().sort((a, b) => {
```

Comment:
why do you use `slice`?
```suggestion
  return [...students].sort((a, b) => {
```

Reply:
but slice() is one way to copy array taught in the course video ;P